cpp/src/arrow/dataset/rados.h

```diff
@@ -0,0 +1,132 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <rados/librados.hpp>
```

Comment:
Ideally this would be included from the `.cc` file any any instance to `ceph::bufferlist` would be replaced by Arrow buffers or arrays.  This helps keep rados out of the Arrow public API (even if the Arrow dist was built with RADOS support).  For example, this is done with our compression utilities and filesystem utilities.  Internal headers can be used if you need templates (though I don't see any here).